x/twap/store.go

```diff
@@ -173,40 +168,26 @@ func (k Keeper) getRecordAtOrBeforeTime(ctx sdk.Context, poolId uint64, t time.T
 		return types.TwapRecord{}, err
 	}
 	store := ctx.KVStore(k.storeKey)
-	// We make an iteration from time=t + 1ns, to time=0 for this pool.
-	// Note that we cannot get any time entries from t + 1ns, as the key would be `prefix|t+1ns`
-	// and the end for a reverse iterator is exclusive. Thus the largest key that can be returned
-	// begins with a prefix of `prefix|t`
-	startKey := types.FormatHistoricalPoolIndexTimePrefix(poolId, time.Unix(0, 0))
-	endKey := types.FormatHistoricalPoolIndexTimePrefix(poolId, t.Add(time.Nanosecond))
```

Comment:
@mattverse this is the +1 ns hack.

This is fixed in this PR, by managing the suffix better, in `FormatHistoricalPoolIndexTimeSuffix`